lld/MachO/Driver.cpp

```diff
@@ -349,7 +349,14 @@ static InputFile *addFile(StringRef path, LoadType loadType,
         Error e = Error::success();
         for (const object::Archive::Child &c : file->getArchive().children(e)) {
           Expected<MemoryBufferRef> mb = c.getMemoryBufferRef();
-          if (!mb || !hasObjCSection(*mb))
+          if (!mb) {
+            // Thin archives from repro tarballs can contain missing members
+            // if the member was not loaded later during the initial link.
```

Comment:
Should we add a `if (c.getParent()->isThin())` here to not drop valid errors?

But outside or repro files, .o files referenced from an archive being missing seems like something we probably want to diag on, hmm.

Maybe we should eagerly put all .o files referenced from thin archives into the repro file always, not just when they're referenced?

Reply:
Not crashing on this seems nice too though, so maybe in addition to adding all .o files from a thin archive to repro files, it'd be nice to also consumeError() here if isThin(), but emit a diag instead of doing it silently.

That seems like a good approach to me.

Reply:
I ended up basically going for the approach outlined here:
- also include unused object files in the repro tarball
- warn on missing member
- make warning toggleable so we don't get spam on stderr when processing old repro files

My last question is what should happen if we don't find members when force-loading, i.e.:

https://github.com/llvm/llvm-project/blob/2ad7b4af95bc333d8f216915cd1b9d688590dcc5/lld/MachO/Driver.cpp#L333-L335

Downgrade this to a warning too? This was not affected by the issue we're fixing (since `fetch()` was called unconditionally here), but if we add a user-facing option, we should be consistent about our behavior.

Reply:
That makes sense to me, yes.